### ISSUE TITLE
[web] enabling semantics breaks scrolling

### ISSUE THREAD

**Issue by tauu** [flutter/flutter#159680]
### Steps to reproduce

Create a page containing two nested widgets that can be scrollable. E.g. a SingleChildScrollView containing a TextField, or TabView containing a SingleChildScrollView. The nested Widget does not have to  be actually scrollable. E.g. it could be a single line TextField or a ListView using NeverScrollPhysics and the issue would still occur.

Enable the Semantics tree.

Run the app in a browser.

Scroll the outer widget.

### Expected results

The outer scrollable Widget should smoothly scroll from top to bottom.

### Actual results

When scrolling over the nested Widget, the scroll offset suddenly jumps to a new position, skipping a substantial amount of the scrollable area.

If the semantics tree is disabled, scrolling works as expected. Surprisingly the jumps also occur even if the nested widget is not actually scrollable. For example a TextField with a fixed amount of lines will also trigger the jumping behaviour wehen scrolling over it.

The same issue was also encountered for a TabView with a nested SingleChildScrollView and a SingleChildScrollView containing a ListView (event when setting scrollPhysics to NeverScrollPhysics).

### Code sample

For a minimal example of the issue, please see https://dartpad.dev/?id=7339cf7cdd671341c70b59356571e949

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/d97769ca-41a9-41a5-95b0-49c5a0948a77

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.24.4, on macOS 15.0.1 24A348 darwin-arm64, locale de-DE)
    • Flutter version 3.24.4 on channel stable at /Applications/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 603104015d (6 weeks ago), 2024-10-24 08:01:25 -0700
    • Engine revision db49896cf2
    • Dart version 3.5.4
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/tauu/Library/Android/sdk
    • Platform android-34, build-tools 33.0.0
    • ANDROID_HOME = /Users/tauu/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/tauu/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 16.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16B40
    ✗ Unable to get list of installed Simulator runtimes.
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.94.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (4 available)
    • iPhone von Georg (mobile)       • 00008030-0019742934DA802E • ios            • iOS 18.0.1 22A3370
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 15.0.1 24A348 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 15.0.1 24A348 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 131.0.6778.86
    ! Error: Browsing on the local area network for Apple Watch von Georg. Ensure the device is unlocked and discoverable via Bluetooth. (code -27)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. Seeing the same behavior as reported on latest sdk versions, although this seems to be specific to web only, as I tried on macOS desktop, and it doesn't replicate on it.

```
stable : 3.24.5
master : 3.27.0-1.0.pre.682
```

**Comment by tauu** [comment]
Indeed, it seems like it only affects the web platform. Maybe there is some interaction with the semantics node for the nested Widget, that causes the issue. For the provided example, I just tried setting the visibility property of the semantics node to hidden and this fixes the scrolling issues. Unfortunately it also eliminates the benefits of the semantics node (e.g. copy/paste is missing in the context menu, if the node is hidden).

<img width="1161" alt="Bildschirmfoto 2024-12-03 um 13 36 36" src="https://github.com/user-attachments/assets/b9bd380b-2920-4438-a943-be98c9015ea2">

**Comment by tauu** [comment]
A workaround for the issue seems to be to just wrap the inner Widget in a GestureDetector that does ... nothing. E.g.:
```dart
SingleChildScrollView(
  child: GestureDetector(
    behavior: HitTestBehavior.translucent,
    onTapDown: (event) { },
    onPanUpdate: (update) { }, 
    child: ...
  ),
)
```
Setting the two handlers is required (even though they don't do anything). Their effects are:
* Setting the `onTapDown` handler will create a semantics node for the `GestureDetector` (setting only the `onPanUpdate` handler does not create a semantics node). 
* Setting the `onPanUpdate` handler will add the style `touch-action: none` and `overflow-y: scroll` to the semantics node for the GestureDetector. 

At least for the provided example this workaround resolves the issue. 


In addition, manually adding the style `touch-action: none` to the semantics node for the TextField in the provided example (this is flt-semantic-node-12 in the screenshot in the previous comment), sometimes also helps reduces the scrolling jumps and sometimes it has no effect. The GestureDetector works consistently.

**Comment by yjbanov** [comment]
I suspect the issue is that the engine fails to detect smooth scrolling and issues a `SemantcsAction.scrollUp` event, which scrolls 80% of the visible area in one frame.

**Comment by tauu** [comment]
@yjbanov your suspicion seems to be correct. Just adding a console.log before `SemanticsAction.scrollUp` / `.scrollDown` is issued in `_recomputeScrollPosition` reveals that whenever the scrolling jumps to a new position  a `SemanticsAction.scrollUp` / `.scrollDown` is issued.

https://github.com/user-attachments/assets/88f2803d-fe91-4fd5-aa19-cef6d7aef782

If the code issuing the `SemanticsAction.scrollUp` / `.scrollDown` is simply removed the scrolling behaviour is once again as expected and the bug is fixed. But I assume these actions are there for a reason 😅 

Do you maybe have a pointer to how this bug should be properly fixed? Some guidance would be much appreciated and I would be happy to submit a PR with a proper fix. :-)

**Comment by Piinks** [comment]
From similar issue https://github.com/flutter/flutter/issues/159358, this looks like a regression introduced in https://github.com/flutter/engine/pull/53922

**Comment by pedromassango** [comment]
@tauu where you able to find a more simpler workaround other than having to add a `GestureDetector` to all my scrollable widgets?

**Comment by tauu** [comment]
ah I see :-(  ... one option would also be to use a custom flutter engine and comment out the lines in the _recomputeScrollPosition function of the engine, that issue a SemanticsAction.scrollUp / .scrollDown. That removed the issue for me, but I don't actually know what the full consequences of removing these lines are.

**Comment by pedromassango** [comment]
Hi @Piinks, any hints to point us to the right direction?

**Comment by Piinks** [comment]
I think this regression from the web team was resolved, the linked issue was closed. @yjbanov this is fixed now, correct?

**Comment by tauu** [comment]
As the fix was only in the engine, could running with a specific engine version (e.g. from beta channel) also be a workaround? According to  https://github.com/flutter/flutter/issues/163896 there is an environment variable to set a specific engine version:  FLUTTER_PREBUILT_ENGINE_VERSION . But I haven't investigated further if that would actually work.

**Comment by pedromassango** [comment]
This is fixed on latest stable, can you confirm @tauu ?
